### PR TITLE
matching: refactor the local control flow of the Unused exception

### PR DESCRIPTION
**Pull request by gasche**
This is a very simple, one-commit refactoring PR that comes out of trying to understand the usage of the `Unused` exception in the pattern-matching compiler. This PR clarifies the exception flow in the intermediate constructs that propagate the exception.

Anyone should feel free to have a look at it, reviewing it requires no specific knowledge of pattern-matching.

### CODE REVIEW

File: lambda/matching.ml [hunk 1]
```diff
@@ -2913,16 +2913,15 @@ let compile_list compile_fun division =
     | (key, cell) :: rem -> (
         if Context.is_empty cell.ctx then
           c_rec totals rem
-        else
-          try
-            let lambda1, total1 = compile_fun cell.ctx cell.pm in
+        else match compile_fun cell.ctx cell.pm with
```
Comment (trefis) [inline lambda/matching.ml]: Could you break the line after `else`?

File: lambda/matching.ml [hunk 2]
```diff
@@ -3027,27 +3027,25 @@ let rec comp_match_handlers comp_fun partial ctx first_match next_matchs =
...
-                  comp_fun
-                    ( match rem with
-                    | [] -> partial
-                    | _ -> Partial
-                    )
-                    ctx_i pm
-                in
+              let partial = match rem with
+                | [] -> partial
+                | _ -> Partial in
```
Comment (trefis) [inline lambda/matching.ml]: Could you break the line before `in`?

**Review by trefis — COMMENTED** [review]
I haven't done a proper review of this (and I'm not going to do one today, so anyone should feel free to jump in).
With that said: humanl-ocamlformat-mode engaged.

File: lambda/matching.ml (hunk 1 above)
Comment (gasche) [reply lambda/matching.ml]: Done.

File: lambda/matching.ml (hunk 2 above)
Comment (gasche) [reply lambda/matching.ml]: Done.

**Review by Octachron — APPROVED** [review]
The PR keeps the existing behavior and narrowing the region where an exception is expected to arise is a win in my book.

**Comment by gasche** [comment]
Thanks! Merging.

**Pull request merged**